### PR TITLE
Add symbolic link to latest run

### PR DESCRIPTION
**Pull request by jprinet**
This will help to simplify automation around the scripts, not having to deal with a randomly named experiment directory

Signed-off-by: Jerome Prinet <jprinet@gradle.com>

### CODE REVIEW

File: components/scripts/lib/init.sh [hunk 1]
```diff
@@ -6,6 +6,13 @@ make_experiment_dir() {
   rm -f "${BUILD_SCAN_FILE}"
 }
 
+create_alias_latest() {
```
Comment (jthurne) [inline components/scripts/lib/init.sh:9]: I suggest renaming this to `make_symlink_to_latest_experiment_dir`

File: components/scripts/gradle/01-validate-incremental-building.sh [hunk 2]
```diff
@@ -64,6 +64,8 @@ execute() {
 
   print_bl
   print_summary
+
+  create_alias_latest
```
[the remark this answers is not in the document]
Comment (jthurne) [reply components/scripts/gradle/01-validate-incremental-building.sh:68]: One more reason to move where this is called: if we do it at the start, then if the script terminates prematurely for some reason, the symlink can be followed to help troubleshoot why the script failed early.

[the remark this answers is not in the document]
Comment (jprinet) [reply components/scripts/gradle/01-validate-incremental-building.sh:68]: This is the exact reason why I decided to put it at the end, if the script fails you may conversely prefer to have `latest` pointing to the last successful experiment...tastes and colors 😅
I'm keen moving it to the beginning if you favor this approach though, just tell me.

If not, I'll move the call to `main` which definitely makes sense.

**Comment by jprinet** [comment]
My last remaining concern @jthurne is about the behavior on Windows OS. 
I added some error redirection to avoid error messages, the link wouldn't be created as a worse case scenario.

**Comment by jthurne** [comment]
> My last remaining concern @jthurne is about the behavior on Windows OS.

Currently, we only support running the scripts using the Windows Subsystem for Linux (WSL), which does support symbolic links.  So we should be good.

> I added some error redirection to avoid error messages, the link wouldn't be created as a worse case scenario.

LGTM.

**Review by jthurne — APPROVED** [review]

**Comment by jthurne** [comment]
@jprinet don't forget to update `release/changes.md`.

**Comment by jprinet** [comment]
> @jprinet don't forget to update `release/changes.md`.

Just done, please confirm you're fine with the wording 👍

File: release/changes.md [hunk 3]
```diff
@@ -1 +1,2 @@
 * FIX: Gradle experiment scripts fail when using the ```--project-dir``` argument to specify the build invocation directory.
+* Add a symbolic link to latest experiment to ease folder navigation
```
Comment (jthurne) [inline release/changes.md:2]:
```suggestion
* Add a symbolic link to latest experiment folder to ease folder navigation.
```

**Review by jthurne — APPROVED** [review]

**Pull request merged**